src/core/request.js

```diff
@@ -44,13 +44,15 @@ Request.prototype.request = function(servers, callback_) {
   const callback = typeof servers === 'function' ? servers : callback_;
 
   this.emit('before');
+
+  const wasRequested = this.requested;
```

Comment:
Maybe user wants another callback added? From reading previous code I've suggested that logic.